### PR TITLE
test: correct wallet_tests

### PR DESCRIPTION
**Pull request by SmartArray**
# DigiByte specific wallet_tests

### What is the intention of this PR?
This PR fixes the `wallet_tests` with DigiByte's specific Wallet Implementation.

### Description of the changes
The commit https://github.com/DigiByte-Core/digibyte/commit/e0ee6755143f569b84e879549a92ab24878a5233 fixes the wallet tests by correcting the test and changing the DigiByte specific parameters.

### How to verify?
1) Change to the directory src/test
2) Compile the test suite
3) Run it using
```bash
./test_digibyte --run_test=wallet_tests
```

### Expected outcome
```
$ ./test_digibyte --run_test=wallet_tests
Running 8 test cases...
*** No errors detected
```

### CODE REVIEW

**Comment by SmartArray** [comment]
@gto90 I closed the previous PR. This PR is much more clean and I tested it once more to make sure nothing is wrong.

Furthermore, I added some comments as previously requested by you here:
https://github.com/DigiByte-Core/digibyte/blob/e0ee6755143f569b84e879549a92ab24878a5233/src/test/test_digibyte.cpp#L168-L177

File: src/test/test_digibyte.cpp [hunk 1]
```diff
@@ -150,20 +150,30 @@ TestChain100Setup::CreateAndProcessBlock(const std::vector<CMutableTransaction>&
...
+        RegenerateCommitments(block, chainActive.Tip(), chainparams.GetConsensus());
         IncrementExtraNonce(&block, chainActive.Tip(), extraNonce);
     }
 
     while (!CheckProofOfWork(GetPoWAlgoHash(block), block.nBits, chainparams.GetConsensus())) ++block.nNonce;
 
     std::shared_ptr<const CBlock> shared_pblock = std::make_shared<const CBlock>(block);
     if (!ProcessNewBlock(chainparams, shared_pblock, true, nullptr)) {
-        throw new std::runtime_error("Could not process new test block");
+        // Uncomment this to catch potential errors. Use lldb/gdb to break on std::runtime_error.
```
Comment (gto90) [inline src/test/test_digibyte.cpp:169]: Thanks for adding this rationale @SmartArray !

**Review by gto90 — APPROVED** [review]
utACK

**Review by digicontributer — APPROVED** [review]
ACK

**Pull request merged**